lib/prism/parse_result.rb

```diff
@@ -5,6 +5,11 @@ module Prism
   # conjunction with locations to allow them to resolve line numbers and source
   # ranges.
   class Source
+    # Create a new source object with the given source code.
+    def self.create(source, start_line = 1, offsets = [])
```

Comment:
How about we override `new` instead, and leave the callsites as they are? That should make it transparent to the consumer.

Reply:
If we override `new` here, won't `rb_class_new_instance` end up performing the ascii only check twice? Once in C and the other one in Ruby?

Reply:
`rb_class_new_instance` doesn't call `::new`, it calls `allocate` and `initialize` directly.